x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/plugin/CanMatchIT.java

```diff
@@ -255,15 +253,53 @@ public void testFailOnUnavailableShards() throws Exception {
         }
     }
 
+    public void testSkipOnNonExistingField() {
+        assertAcked(client().admin().indices().prepareCreate("index1").setMapping("filed1", "type=keyword"));
+        client().prepareBulk("index1")
+            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+            .add(new IndexRequest().source("field1", "a"))
+            .get();
+
+        assertAcked(client().admin().indices().prepareCreate("index2").setMapping("filed2", "type=keyword"));
+        client().prepareBulk("index2")
+            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+            .add(new IndexRequest().source("field2", "b"))
+            .get();
+
+        try {
+            Set<String> queriedIndices = ConcurrentCollections.newConcurrentSet();
+            for (var transportService : internalCluster().getInstances(TransportService.class)) {
+                as(transportService, MockTransportService.class).addRequestHandlingBehavior(
+                    ComputeService.DATA_ACTION_NAME,
+                    (handler, request, channel, task) -> {
+                        DataNodeRequest dataNodeRequest = (DataNodeRequest) request;
+                        for (ShardId shardId : dataNodeRequest.shardIds()) {
+                            queriedIndices.add(shardId.getIndexName());
+                        }
+                        handler.messageReceived(request, channel, task);
+                    }
+                );
+
+            }
+            try (var resp = run("from index* | WHERE field1 == \"a\"")) {
+                assertThat(getValuesList(resp), hasSize(1));
+                assertThat(queriedIndices, equalTo(Set.of("index1")));
```

Comment:
This is currently failing as request is send to both indices even though `index2` does not have `field1`